pennylane/devices/qubit/measure.py

```diff
@@ -180,7 +180,7 @@ def get_measurement_function(
         if isinstance(measurementprocess.mv, MeasurementValue):
             return state_diagonalizing_gates
 
-        if isinstance(measurementprocess, ExpectationMP):
+        if isinstance(measurementprocess, ExpectationMP) and measurementprocess.obs is not None:
```

Comment:
What's producing these now?

Reply:
`MP(mv)` gets replaced by `MP(eigvals, wires)` by `defer_measurements`. Sorry, I should've specified in the PR description. Updated now.